src/backend/commands/tablecmds.c

```diff
@@ -16058,10 +16058,11 @@ wack_pid_relname(AlterPartitionId 		 *pid,
 
 		par_prule = *ppar_prule;
 
-		*plrelname = par_prule->relname;
-
 		if (par_prule && par_prule->topRule && par_prule->topRule->children)
```

Comment:
I think you should have just replaced the `par_prule &&` condition with an assert.

Reply:
Wait, how that would help with assert works on debug builds only?

Reply:
You should remove condition and add assert

Reply:
Can you explain why I should add assert if the analyzer doesn't see it?

Reply:
for testing in debug build on CI

Reply:
But that wouldn't fix the problem with the analyzer

Reply:
problem with the analyzer is fixed by removing condition

Reply:
But you told earlier that using assert is not a solution. When I should use Insist and when I should use assert?

Reply:
in this case you may remove condition and add assert
in another case you may add insist, i.e. add new condition

Reply:
Done